spec/mongoid/criteria/queryable/selectable_spec.rb

```diff
@@ -3385,16 +3385,34 @@ class TestModel
         end
 
         it "constructs a text search document" do
-          expect(selection.selector[:$text][:$search]).to eq("essais")
+          expect(selection.selector['$text']['$search']).to eq("essais")
         end
 
         it "add the options to the text search document" do
-          expect(selection.selector[:$text][:$language]).to eq("fr")
+          expect(selection.selector['$text'][:$language]).to eq("fr")
         end
 
         it_behaves_like "a cloning selection"
       end
     end
+
+    context 'when given more than once' do
+      let(:selection) do
+        query.text_search("one").text_search('two')
+      end
+
+      # MongoDB server can only handle one text expression at a time,
+      # per https://docs.mongodb.com/manual/reference/operator/query/text/.
+      # Nonetheless we test that the query is built correctly when
+      # a user supplies more than one text condition.
+      it 'merges conditions' do
+        expect(selection.selector).to eq('$and' => [
```

Comment:
Should we also test that the warning occurs?

Reply:
I do not consider this undertaking worthwhile.

Reply:
Why not? We do it in many other places in both the driver and mongoid, and it only requires an extra line or two of code

Reply:
Because I am satisfied with the current implementation and would like to move on to other things. Feel free to file a ticket to test the warning if you find this worthwhile.